framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -452,6 +459,52 @@ public void updateInferredMethodReturnType(
         AnnotatedTypeMirror rhsATM = atf.getAnnotatedType(retNode.getTree().getExpression());
         helper.updateAnnotationSetInScene(
                 method.returnType, atf, jaifPath, rhsATM, lhsATM, TypeUseLocation.RETURN);
+
+        // Now, update overridden methods based on the implementation we just saw.  This inference
+        // is similar to the inference procedure for method parameters: both are updated based only
+        // on the implementations (in this case) or call-sites (for method parameters) that are
+        // available to WPI.
+        //
+        // TODO: Performing this for each return statement is inefficient. It would
+        //  be enough to propagate the information once at the end of WPI, for the
+        //  final return type.
+        //
+        // In addition, it's inefficient to propagate (elsewhere) when the overriding
+        // relationship is discovered. More information may be discovered but not propagated.
+        // This could lead to an extra iteration of WPI, which would be very costly.
+        //
+        // It would be better to record overriding relationships as they are discovered,
```

Comment:
Rather than writing a TODO comment, could you just implement this change?  It's true that fixpoint iteration is expensive, but anything we can do to avoid a extra iteration is valuable.

Reply:
Mike and I agreed that the approach described in this todo comment is no more efficient, and possibly less efficient, than the approach I actually implemented; neither can lead to an extra WPI pass (since the values aren't used until the next WPI pass anyway), and the approach Mike suggested requires another AST pass.